File: src/2042.check-if-numbers-are-ascending-in-a-sentence.2168/solution.ts
function areNumbersAscending (s: string): boolean {
    const nums = s.split(' ').filter((item) => /^\d+$/.test(item)).map(item => +item);
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] <= nums[i - 1]) {
            return false;
        }
    }
    return true;
}

File: src/2042.check-if-numbers-are-ascending-in-a-sentence.2168/solution.java
class Solution {
    public boolean areNumbersAscending(String s) {
        int lasNum = 0;
        String[] ss = s.split(" ");
        for (String sss : ss) {
            if (Character.isDigit(sss.charAt(0))) {
                int n = Integer.parseInt(sss);
                if (n <= lasNum) {
                    return false;
                }
                lasNum = n;
            }
        }
        return true;
    }
}